packages/customize-widgets/src/components/sidebar-block-editor/use-sidebar-block-editor.js

```diff
@@ -60,29 +63,25 @@ function widgetToBlock( widget ) {
 	let block;
 
 	if ( widget.idBase === 'block' ) {
-		const parsedBlocks = parse(
-			widget.instance.__unstable_instance.content
-		);
+		const parsedBlocks = parse( widget.instance.raw_instance.content );
 		block = parsedBlocks.length
 			? parsedBlocks[ 0 ]
 			: createBlock( 'core/paragraph', {} );
-	} else {
-		const attributes = {
-			name: widget.name,
-			form: widget.form,
-			instance: widget.instance,
+	} else if ( widget.number ) {
+		// Widget that extends WP_Widget.
+		block = createBlock( 'core/legacy-widget', {
 			idBase: widget.idBase,
-			number: widget.number,
-		};
-
-		const isReferenceWidget = ! widget.widgetClass;
-		if ( isReferenceWidget ) {
-			attributes.referenceWidgetName = widget.id;
-		} else {
-			attributes.widgetClass = widget.widgetClass;
-		}
-
-		block = createBlock( 'core/legacy-widget', attributes, [] );
+			instance: {
```

Comment:
this function is small and filled with long paths to props. How about cleaning the function's body a little by introducing destructuring the function's argument?

```js
function widgetToBlock({id, idBase, number, instance: {
  raw_instance: raw,
  encoded_serialized_instance: encoded,
  instance_hash_key: hash,
} ) 
```

it will make the entire function more readable, with places like this, simplified to:

```js
block = createBlock( 'core/legacy-widget', {
   idBase,
   instance: {
      encoded,
      hash,
      raw,
},
```
or even to 
```js
block = createBlock( 'core/legacy-widget', { idBase, instance } );
```
if we need to assign all props

the same idea can be applied to other functions as well, to avoid places like in line 35.
Probably in the separate PR.

Reply:
👍 f7d4896427072a064cf260dbcb0c57b16b82794a